server/src/main/java/org/jetbrains/maven/server/MavenBSPServer.java

```diff
@@ -170,7 +174,13 @@ public void onBuildExit() {
 
     @Override
     public CompletableFuture<CompileResult> buildTargetCompile(CompileParams compileParams) {
-        return CompletableFuture.completedFuture(new CompileResult(StatusCode.OK));
+        MavenController controller = new MavenController(client);
```

Comment:
Is it necessary to initialize controller for each build target request?

Reply:
Not really. but where could we construct MavenController?